File: daily/array_matrix/274.cpp
#include "header.h"
//H指数
class Solution {
public:
    int hIndex(vector<int>& citations) {
        sort(citations.begin(), citations.end());
        int n = citations.size() - 1;
        int h = 0;
        while (n >= 0 && citations[n] > h) {
            ++h;
            --n;
        }
        return h;
    }
};

File: daily/array_matrix/274.py
from typing import List
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        citations.sort()
        h = 0
        i = len(citations) - 1
        while i >= 0 and citations[i] > h :
            h += 1
            i -= 1
        return h

